scheduler/task.go

```diff
@@ -42,6 +42,57 @@ func createTaskInfo(task types.EremeticTask, offer *mesos.Offer) (types.Eremetic
 		})
 	}
 
+	var portResources []*mesos.Value_Range
+	var portMapping   []*mesos.ContainerInfo_DockerInfo_PortMapping
+
+	if(len(task.Ports) > 0) {
```

Comment:
The problem I had was with C... if it's not done in the same context as A it may lead to non-overlapping ranges (ie: you get offer (3001:3003),(3007:3010) and you need 5 ports, the list will be [3001, 2, 3, 7, 8] and the resulting range (3001, 3008) which will violate the offer)